### ISSUE TITLE
In supervised learning mode, how to handle multi-label tasks? 

### ISSUE THREAD

**Issue by genewlan** [MaartenGr/BERTopic#1414]
It is common for a document to have more than one label, such as a blog post that may have multiple topics. Currently, in  supervised BERTopic mode, I noticed that in the code snippet "topics, _ = topic_model.fit_transform(docs, y=targets)", "y" can only take a list of integers, which means that each document can only have one label. What should I do if I want to input multiple labels for each document? Do you have any suggestions?

**Comment by MaartenGr** [comment]
Multi-label supervised topic modeling is indeed currently not supported. What you can do, however, is to approach that task by first generating representations of topics and then splitting up the documents with [`.approximate_distribution`](https://maartengr.github.io/BERTopic/getting_started/distribution/distribution.html) to get multiple labels for each document. Your best bet might be to use `use_embedding_model=True` as that will more accurately represent the initial training process. 

In other words, you can directly train on your data and then use `.approximate_distribution` to get a more fine-grained representation per document.

**Comment by genewlan** [comment]
BERTopic is a topic modeling tool based on a single topic model, which can only recognize and extract one set of topics. If I want to obtain multiple sets of topics, is it reasonable to solve it by training multiple models with different parameters，as fllow： 
1、Generate multiple sets of topics using models with different parameters.
2、Merge these topics based on their similarity to obtain a composite topic.

**Comment by MaartenGr** [comment]
Sure, that seems reasonable. What you can do it generate these multiple sets of topics, merge those based on their similarity and a chosen threshold, and finally give them back to BERTopic through [manual topic modeling](https://maartengr.github.io/BERTopic/getting_started/manual/manual.html). That will give you a single topic model that holds all of the topics you created with your other topic models.
